Description: Remove non-determinism from build process
 .
Author: Balint Reczey <balint@balintreczey.hu>
Forwarded: not-needed
--- a/src/libcec/CMakeLists.txt
+++ b/src/libcec/CMakeLists.txt
@@ -129,7 +129,7 @@
 source_group("Header Files" FILES ${CEC_HEADERS})
 
 # platform and device specific
-include(cmake/SetBuildInfo.cmake)
+set(LIB_INFO "compiled on ${CMAKE_SYSTEM} ... ")
 include(cmake/CheckPlatformSupport.cmake)
 
 ## create project groups
