--- a/tools/depends/target/ffmpeg/CMakeLists.txt
+++ b/tools/depends/target/ffmpeg/CMakeLists.txt
@@ -17,7 +17,7 @@ if(CROSSCOMPILING)
   list(APPEND ffmpeg_conf --ar=${CMAKE_AR} --strip=${CMAKE_STRIP})
   message(STATUS "CROSS: ${ffmpeg_conf}")
 endif()
-
+list(APPEND ffmpeg_conf --cpu=cortex-a73.cortex-a53 --arch=arm)
 if(CMAKE_C_FLAGS)
   list(APPEND ffmpeg_conf --extra-cflags=${CMAKE_C_FLAGS})
 endif()
