Index: kodi/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
===================================================================
--- kodi.orig/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
+++ kodi/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
@@ -101,12 +101,6 @@ bool CDVDVideoCodecAmlogic::Open(CDVDStr
   if (hints.stills || hints.width == 0)
     return false;
 
-  if (!aml_permissions())
-  {
-    CLog::Log(LOGERROR, "AML: no proper permission, please contact the device vendor. Skipping codec...");
-    return false;
-  }
-
   // allow only 1 instance here
   if (m_InstanceGuard.exchange(true))
   {
Index: kodi/xbmc/utils/AMLUtils.cpp
===================================================================
--- kodi.orig/xbmc/utils/AMLUtils.cpp
+++ kodi/xbmc/utils/AMLUtils.cpp
@@ -42,106 +42,6 @@ bool aml_present()
   return has_aml == 1;
 }
 
-bool aml_permissions()
-{
-  if (!aml_present())
-    return false;
-
-  static int permissions_ok = -1;
-  if (permissions_ok == -1)
-  {
-    permissions_ok = 1;
-
-    if (!SysfsUtils::HasRW("/dev/amvideo"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /dev/amvideo");
-      permissions_ok = 0;
-    }
-    if (!SysfsUtils::HasRW("/dev/amstream_mpts"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /dev/amstream*");
-      permissions_ok = 0;
-    }
-    if (!SysfsUtils::HasRW("/sys/class/video/axis"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/video/axis");
-      permissions_ok = 0;
-    }
-    if (!SysfsUtils::HasRW("/sys/class/video/screen_mode"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/video/screen_mode");
-      permissions_ok = 0;
-    }
-    if (!SysfsUtils::HasRW("/sys/class/video/disable_video"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/video/disable_video");
-      permissions_ok = 0;
-    }
-    if (!SysfsUtils::HasRW("/sys/class/tsync/pts_pcrscr"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/tsync/pts_pcrscr");
-      permissions_ok = 0;
-    }
-    if (!SysfsUtils::HasRW("/dev/video10"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /dev/video10");
-      permissions_ok = 0;
-    }
-    if (!SysfsUtils::HasRW("/sys/module/ionvideo/parameters/freerun_mode"))	    
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/module/ionvideo/parameters/freerun_mode");
-      permissions_ok = 0;
-    }
-    if (!SysfsUtils::HasRW("/sys/class/video/freerun_mode"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/video/freerun_mode");
-      permissions_ok = 0;
-    }
-    if (!SysfsUtils::HasRW("/sys/class/audiodsp/digital_raw"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/audiodsp/digital_raw");
-    }
-    if (!SysfsUtils::HasRW("/sys/class/amhdmitx/amhdmitx0/config"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/amhdmitx/amhdmitx0/config");
-    }
-    if (!SysfsUtils::HasRW("/sys/class/vfm/map"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/vfm/map");
-    }
-    if (!SysfsUtils::HasRW("/sys/class/tsync/enable"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/tsync/enable");
-    }
-    if (!SysfsUtils::HasRW("/sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq");
-    }
-    if (!SysfsUtils::HasRW("/sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq");
-    }
-    if (!SysfsUtils::HasRW("/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor");
-    }
-    if (aml_has_frac_rate_policy() && !SysfsUtils::HasRW("/sys/class/amhdmitx/amhdmitx0/frac_rate_policy"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/amhdmitx/amhdmitx0/frac_rate_policy");
-    }
-    if (!SysfsUtils::HasRW("/sys/module/di/parameters/bypass_prog"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/module/di/parameters/bypass_prog");
-    }
-    if (!SysfsUtils::HasRW("/sys/class/display/mode"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/class/display/mode");
-    }
-  }
-
-  return permissions_ok == 1;
-}
-
 bool aml_support_hevc()
 {
   static int has_hevc = -1;
Index: kodi/xbmc/utils/AMLUtils.h
===================================================================
--- kodi.orig/xbmc/utils/AMLUtils.h
+++ kodi/xbmc/utils/AMLUtils.h
@@ -42,7 +42,6 @@ enum AML_SUPPORT_H264_4K2K
 };
 
 bool aml_present();
-bool aml_permissions();
 bool aml_support_hevc();
 bool aml_support_hevc_4k2k();
 bool aml_support_hevc_10bit();
Index: kodi/xbmc/windowing/amlogic/WinSystemAmlogic.cpp
===================================================================
--- kodi.orig/xbmc/windowing/amlogic/WinSystemAmlogic.cpp
+++ kodi/xbmc/windowing/amlogic/WinSystemAmlogic.cpp
@@ -62,7 +62,6 @@ CWinSystemAmlogic::CWinSystemAmlogic() :
   m_stereo_mode = RENDER_STEREO_MODE_OFF;
   m_delayDispReset = false;
 
-  aml_permissions();
   aml_disable_freeScale();
 
   // Register sink
