From 1fdb904c496738bcc2f228741b71ec2792d97112 Mon Sep 17 00:00:00 2001
From: cdu13a <cdu13a@gmail.com>
Date: Sun, 27 Jan 2019 01:31:31 -0500
Subject: [PATCH] RetroPlayer alpha channel fix

---
 xbmc/utils/AMLUtils.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index a03b58b5783..1ec659c164b 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -672,8 +672,8 @@ bool aml_set_reg_ignore_alpha()
 {
   if (aml_has_capability_ignore_alpha())
   {
-    std::string path = "/sys/kernel/debug/aml_reg/paddr";
-    if (SysfsUtils::SetString(path, "d01068b4 0x7fc0") == 0)
+    std::string path = "/sys/class/graphics/fb0/debug";
+    if (SysfsUtils::SetString(path, "write 0x1a2d 0x7fc0") == 0)
       return true;
   }
   return false;
@@ -683,8 +683,8 @@ bool aml_unset_reg_ignore_alpha()
 {
   if (aml_has_capability_ignore_alpha())
   {
-    std::string path = "/sys/kernel/debug/aml_reg/paddr";
-    if (SysfsUtils::SetString(path, "d01068b4 0x3fc0") == 0)
+    std::string path = "/sys/class/graphics/fb0/debug";
+    if (SysfsUtils::SetString(path, "write 0x1a2d 0x3fc0") == 0)
       return true;
   }
   return false;
